Fedora 23 support branch

jgriffiths opened the pull request:
This is a combination of the changes from @marmarek for Fedora 22 plus some further fixes to support Fedora 23.

This is enough to get installer-qubes-os building and fix a few issues building the .iso, but there remains an issue with lvm2 requiring system-release >= 23 which this patch-set doesn't address yet.

Mainly raising a PR for tracking purposes, no need to review or think about merging until a workaround for lvm2 is found and make iso is fully working.

marmarek committed:
pykickstart: update for Fedora 21

marmarek committed:
pungi: update for Fedora 21

marmarek committed:
lorax: update for Fedora 21 - part 1

marmarek committed:
anaconda: update to 21.48.21-1

Apply diff anaconda-20.25.16-1..anaconda-21.48.21-1

marmarek committed:
anaconda: update Qubes-specific code for Fedora 21 version

marmarek committed:
comps: update based on Fedora 21 version

marmarek committed:
anaconda: lower disk requirements

Base installed system, including one template uses about 8GB. So make it
possible to install the system with root partition about that size. This
is probably bad idea, but the limit here is hard - it isn't possible to
force installation on smaller partition.

marmarek committed:
Update makefile to handle ISO filename generated by new anaconda

marmarek committed:
pykickstart: fix repo --gpgkey option patch

marmarek committed:
lorax: keep GTK theme installed, minor cleanup

marmarek committed:
anaconda: allow locked root password

marmarek committed:
anaconda: update to 22.20.13-1

Apply diff anaconda-21.48.21-1..anaconda-22.20.13-1

marmarek committed:
pykickstart: update to 1.99.66

marmarek committed:
pungi: update to 3.14

marmarek committed:
qubes-release: add Fedora 22 release key

marmarek committed:
conf: switch to Fedora 22 repositories

marmarek committed:
lorax: update templates based on lorax-22.11-1.fc22

marmarek committed:
anaconda: remove autogenerated configure scripts

marmarek committed:
lorax: install additional files needed by syslinux-6.x

marmarek committed:
anaconda: use kernel-install instead of grubby to regenerate initrd/grub.conf

Since we have own hook there, it properly handles Xen. This means we no
longer need post scripts in kickstart for that.

Conflicts:
	anaconda/pyanaconda/packaging/__init__.py

jgriffiths committed:
pungi: do not generate upgrade.img

We do not support in-place upgrades this way.

jgriffiths committed:
lorax: remove few more firmware packages

jgriffiths committed:
Remove patch thats been applied upstream

jgriffiths committed:
Remove and gitignore checked in generated file

jgriffiths committed:
Add the Fedora 23 signing key

jgriffiths committed:
conf: switch to Fedora 23 repositories

jgriffiths committed:
Add lorax 22.1

Fedora 23 forces every app that can be run under python 3 to be installed
as a python 3 app. This means lorax is installed as python 3 while pungi
remains 2.7. As soon as pungi is updated to python 3 (or the Fedora team
come to their senses) this package can be deleted.

jgriffiths committed:
pykickstart: Upgrade to 23.1

Forced upgrade to support a split python 2.7/3.4 install.

jgriffiths committed:
Order RPM_SPEC_FILES by dependency

Also remove tabs which should only appear before target actions.

jgriffiths committed:
pykickstart: Document how to get the source

marmarek commented:
> but there remains an issue with lvm2 requiring system-release >= 23 which this patch-set doesn't address yet.

Take a look at `qubes-release` package. It provides `system-release`. Maybe setting its version to 23 will help? But make sure `dnf` will use `qubes-release` version for `$releasever` variable. In `yum` there was a config option named `distroverpkg` to choose the package, not sure about `dnf`.

jgriffiths commented:
> Take a look at qubes-release package. It provides system-release. Maybe setting its version to 23 will help?

I tried this and it didn't help unfortunately. The [linked bug](https://bugzilla.redhat.com/show_bug.cgi?id=1224561) suggests that lvm2 should be fixed since relying on system-release versioning is hostile to other spins (which obviously includes us). But there's been no progress on it and its been closed. More Fedora arrogance :-(

I will investigate further when I get some free time. I really don't want to bring in lvm2 just to disable its system-release dependency. It was bad enough having to add lorax for the python split.

jgriffiths committed:
Provide system-release v23+ for lvm2 dependencies

jgriffiths commented:
> But make sure dnf will use qubes-release version for $releasever variable.

Note: I'm not sure how to make that happen. 

I must have made a typo in my previous munging of the spec version, or something was dirty, after re-implementing (a480c93) and building I get further:

```
Can't stat exclude path "/install", No such file or directory - ignoring.
Error connecting to audit system.
<python stack trace>
subprocess.CalledProcessError: Command '['setfiles', '-e', '/proc', '-e', '/sys', '-e', '/dev', '-e', '/install', '/etc/selinux/targeted/contexts/files/file_contexts', '/']' returned non-zero exit status 255
Makefile:55: recipe for target 'iso-installer' failed
```

edit: This appears to be because I have turned off audit spam in my dev VM kernel options. I had to do this because all the other methods to disable it fail to work. It seems setfile has gained a mandatory dependency on the audit subsystem being enabled. Can't wait for debian in dom0.

I'm out of time for today, will investigate that tomorrow. Getting closer!

jgriffiths committed:
anaconda: Include anaconda-dracut as a dependency

jgriffiths committed:
Add dracut-live to iso deps list and the anaconda install group

This package provides img-lib which is needed by dmsquash and the
anaconda dracut plugin. Seems to be needed in the build chroot as
well as the install iso for 'make iso' to work.

jgriffiths committed:
lorax: Pass the kernel version to dracut as --kver

dracut dies with a usage message here if the version is passed after the
image name.

This is probably not needed as it turns out but I can't test that without a
full clean build which I don't have time for right now.

jgriffiths committed:
anaconda: Add dracut-live as a dependency

jgriffiths committed:
lorax: Don't disable network for efi image creation

Doing so prevents livenet from loading and causes dracut to fail.

jgriffiths committed:
lorax: Don't delete os-release directories

Doing so prevents the install .iso from booting. First identified under
https://bugzilla.redhat.com/show_bug.cgi?id=1191713 and a second fix
was put in at https://github.com/rhinstaller/lorax/pull/6/commits/c0b10633977625e3692397fa428a39d40aeb7772
(this patch includes both changes).

jgriffiths committed:
lorax: Create the qubes directory before calling dracut

marmarek commented on lorax-templates-qubes/templates/efi.tmpl:
```diff
@@ -41,7 +41,7 @@ install boot/efi/EFI/*/fonts/unicode.pf2 ${EFIBOOTDIR}/fonts/
     %if include_kernel:
         copy ${KERNELDIR}/vmlinuz ${EFIBOOTDIR}
-        runcmd chroot ${inroot} dracut --nomdadmconf --nolvmconf --xz --install '/.buildstamp' --add 'anaconda pollcdrom' --omit "network multipath modsign systemd crypt shutdown plymouth fcoe fcoe-uefi nfs iscsi ifcfg" --omit-drivers="${scsi_modules}" --omit-drivers="${extra_modules}" /boot/efi/EFI/qubes/initrd-small.img ${kver}
+        runcmd chroot ${inroot} dracut --nomdadmconf --nolvmconf --xz --install '/.buildstamp' --add 'anaconda pollcdrom' --omit "multipath modsign systemd crypt shutdown plymouth fcoe fcoe-uefi nfs iscsi ifcfg" --omit-drivers="${scsi_modules}" --omit-drivers="${extra_modules}" /boot/efi/EFI/qubes/initrd-small.img ${kver}
```
It is critical to have efiboot.img smaller than 32MB. Otherwise it will not boot (because of [ISO9660 limitation](https://github.com/QubesOS/qubes-issues/issues/794#issuecomment-135988806)).
